source/drivers/DynamicPwm.cpp

```diff
@@ -89,82 +43,9 @@ void gpiote_reinit(PinName pin, PinName oldPin, uint8_t channel_number)
   * @param persistance the level of persistence for this pin PWM_PERSISTENCE_PERSISTENT (can not be replaced until freed, should only be used for system services really.)
   *                    or PWM_PERSISTENCE_TRANSIENT (can be replaced at any point if a channel is required.)
   */
-DynamicPwm::DynamicPwm(PinName pin, PwmPersistence persistence) : PwmOut(pin)
+DynamicPwm::DynamicPwm(PinName pin) : PwmOut(pin)
 {
-    this->flags = persistence;
-}
-
-/**
-  * Redirects the pwm channel to point at a different pin.
-  *
-  * @param pin the desired pin to output a PWM wave.
-  *
-  * @code
-  * DynamicPwm* pwm = DynamicPwm::allocate(PinName n);
-  * pwm->redirect(p0); // pwm is now produced on p0
-  * @endcode
-  */
-void DynamicPwm::redirect(PinName pin)
-{
-    gpiote_reinit(pin, _pwm.pin, (uint8_t)_pwm.pwm);
-    this->_pwm.pin = pin;
-}
-
-/**
-  * Creates a new DynamicPwm instance, or reuses an existing instance that
-  * has a persistence level of PWM_PERSISTENCE_TRANSIENT.
-  *
-  * @param pin the name of the pin for the pwm to target
-  *
-  * @param persistance the level of persistence for this pin PWM_PERSISTENCE_PERSISTENT (can not be replaced until freed, should only be used for system services really.)
-  *                    or PWM_PERSISTENCE_TRANSIENT (can be replaced at any point if a channel is required.)
-  *
-  * @return a pointer to the first available free pwm channel - or the first one that can be reallocated. If
-  *         no channels are available, NULL is returned.
-  *
-  * @code
-  * DynamicPwm* pwm = DynamicPwm::allocate(PinName n);
-  * @endcode
-  */
-DynamicPwm* DynamicPwm::allocate(PinName pin, PwmPersistence persistence)
-{
-    //try to find a blank spot first
-    for(int i = 0; i < NO_PWMS; i++)
-    {
-        if(pwms[i] == NULL)
-        {
-            lastUsed = i;
-            pwms[i] = new DynamicPwm(pin, persistence);
-            return pwms[i];
-        }
-    }
-
-    //no blank spot.. try to find a transient PWM
-    int channelIterator = (lastUsed + 1 > NO_PWMS - 1) ? 0 : lastUsed + 1;
-
-    while(channelIterator != lastUsed)
-    {
-        if(pwms[channelIterator]->flags & PWM_PERSISTENCE_TRANSIENT)
-        {
-            lastUsed = channelIterator;
-            pwms[channelIterator]->flags = persistence;
-            pwms[channelIterator]->redirect(pin);
-            return pwms[channelIterator];
-        }
-
-        channelIterator = (channelIterator + 1 > NO_PWMS - 1) ? 0 : channelIterator + 1;
-    }
-
-    //if we haven't found a free one, we must try to allocate the last used...
-    if(pwms[lastUsed]->flags & PWM_PERSISTENCE_TRANSIENT)
-    {
-        pwms[lastUsed]->flags = persistence;
-        pwms[lastUsed]->redirect(pin);
-        return pwms[lastUsed];
-    }
-
-    //well if we have no transient channels - we can't give any away! :( return null
-    return (DynamicPwm*)NULL;
+    this->period = 0;
```

Comment:
Do we really want 0 as the default? I seem to recall we default to 20ms at the moment? (servo frequency)?